lib/cassia/response_handlers/discover_descriptor_of_char.rb

```diff
@@ -0,0 +1,25 @@
+module Cassia
+  module ResponseHandlers
+    class DiscoverDescriptorOfChar
+      def initialize(access_controller, router: , device_mac: , char_uuid: )
+        @access_controller = access_controller
+        @router = router
+        @device_mac = device_mac
+        @char_uuid = char_uuid
+      end
+
+      def handle(response)
+        device = @router.connected_devices.detect {|device| device.mac == @device_mac}
+        device = Device.new(mac: @device_mac) if device.nil?
```

Comment:
is there a reason why you added the creation of the device and char here but we aren't doing this in the other response handlers?